src/commonMain/kotlin/de/westnordost/osmfeatures/IDLocalizedFeatureCollection.kt

```diff
@@ -83,4 +83,12 @@ private fun String.getLanguageComponents(): Sequence<String> = sequence {
         yield(listOf(language, other).joinToString("-"))
     }
     yield(this@getLanguageComponents)
-}
\ No newline at end of file
+}
+
+/** If a localized feature has no names or no terms and fallbacks are available, apply those. */
+private fun LocalizedFeature.withFallback(fallback: Feature?) =
+    if (fallback == null) this
+    else copy(
+        names = names.ifEmpty { fallback.names },
+        terms = terms.ifEmpty { fallback.terms },
+    )
```

Comment:
Should check first if own names, terms or aliases are empty before copying to avoid copying in the majority of cases.

Reply:
Fair point. I couldn't find an especially elegant way to do it, so let me know if you can think of anything nicer :)